File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
vowel = ['a','e','i','o','u']

while True:
    result = True
    word = input().rstrip('\n')
    if word=='end':
        break
    words = list(word)
    for i in range(1, len(words)):
        if words[i]==words[i-1] and words[i] not in ['e', 'o']:
            result=False
    
    vowel_cnt = 0
    consonant_cnt = 0
    total_vowel_cnt = 0
    for w in words:
        if w in vowel:
            vowel_cnt+=1
            consonant_cnt=0
            total_vowel_cnt+=1
        else:
            consonant_cnt+=1
            vowel_cnt=0
        if vowel_cnt>=3 or consonant_cnt>=3:
            result=False
    if total_vowel_cnt<=0:
        result=False
    if result:
        print("<{0}> is acceptable.".format(word))
    else:
        print("<{0}> is not acceptable.".format(word))

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"slices"
	"strings"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
	vowel  = []string{"a", "e", "i", "o", "u"}
)

func main() {
	defer writer.Flush()
	var s string
	var vowelCnt, consonantCnt, totalVowelCnt = 0, 0, 0
	var result = true
	for {
		vowelCnt, consonantCnt, totalVowelCnt = 0, 0, 0
		result = true
		fmt.Fscanf(reader, "%s\n", &s)

		if s == "end" {
			break
		}

		for i := 0; i < len(vowel); i++ {
			if strings.Contains(s, vowel[i]) {
				totalVowelCnt += 1
			}
		}
		if totalVowelCnt <= 0 {
			result = false
		}
		sArr := strings.Split(s, "")

		for i := 1; i < len(sArr); i++ {
			if (sArr[i-1] == sArr[i]) && (sArr[i] != "o" && sArr[i] != "e") {
				result = false
			}
		}
		for i := 0; i < len(sArr); i++ {
			if slices.Contains(vowel, sArr[i]) {
				vowelCnt += 1
				consonantCnt = 0
			} else {
				vowelCnt = 0
				consonantCnt += 1
			}
			if vowelCnt >= 3 || consonantCnt >= 3 {
				result = false
			}
		}
		if result {
			fmt.Fprintf(writer, "<%s> is acceptable.\n", s)
		} else {
			fmt.Fprintf(writer, "<%s> is not acceptable.\n", s)
		}
	}
}
